app/src/main/java/fr/groggy/racecontrol/tv/RaceControlTvApplication.kt

```diff
@@ -1,14 +1,27 @@
 package fr.groggy.racecontrol.tv
 
 import android.app.Application
+import android.content.res.Configuration
+import android.content.res.Resources
 import com.jakewharton.threetenabp.AndroidThreeTen
 import dagger.hilt.android.HiltAndroidApp
+import fr.groggy.racecontrol.tv.utils.locale.LocaleManager
+import javax.inject.Inject
 
 @HiltAndroidApp
 class RaceControlTvApplication: Application() {
+    @Inject
+    lateinit var localeManager: LocaleManager
+
     override fun onCreate() {
         super.onCreate()
+        localeManager.currentLocale = localeManager.getISO3LanguageFromConfiguration(Resources.getSystem().configuration)
 
         AndroidThreeTen.init(this)
     }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        localeManager.currentLocale = localeManager.getISO3LanguageFromConfiguration(newConfig)
+    }
```

Comment:
Hm...

I'd say hardly we are going to have a configuration change on the TV. If that happens would be at activity level in this case not application.

But you can always access the current locale on the `Configuration`, so let's remove this from the application, and also instead of a need to initiate the `currentLocale` explicitly, let's do it via constructor of the `LocaleManager` class so then it is 100% guaranteed that we always have a locale.